### PR TITLE
Update CI and MCSCF sections of manual

### LINKED ISSUE #600: detci expansion length for truncated CI
It would appear that detci sets up string lengths that correspond to FCI, which makes truncated CI calculations very expensive. For example, (10e,10o) gives
    There are 252 alpha and 252 beta strings
    The CI space requires 63504 (6.35E+04) determinants and 1 blocks
regardless of the truncation level.

What's the correct way to run truncated CI with detci? Should I be assigning the states to RAS I and III and leave RAS II empty?

### PR DESCRIPTION
**Pull request by CDSherrill**
## Description
Update the CI documentation

## Todos
Notable points that this PR has either accomplished or will accomplish.
* **Developer Interest**
  - [x] Split off MCSCF docs to own page
  - [x] Updated CI documentation, closes #600 
  - [x] Fixed typo to address #621 

## Status
- [x]  Ready to go

### CODE REVIEW

File: doc/sphinxman/source/detci.rst [hunk 1]
```diff
@@ -135,6 +128,38 @@ which the driver has built-in shortcuts, so long as the relevant options
...
 in :source:`samples`.
 
+.. _`table:ci_spaces`:
+
+.. table:: Orbital spaces for CI computations
+
+    +----------------------------+----------------------------+----------------------------+
+    | CI (e.g., CISD, FCI)       | RASCI                      | CASCI                      |
+    +============================+============================+============================+
+    | |globals__frozen_uocc|     | |globals__frozen_uocc|     |                            |
```
Comment (dgasmith) [inline doc/sphinxman/source/detci.rst]: Shouldn't `frozen_uocc` be here for CASCI as well?

File: doc/sphinxman/source/detci.rst [hunk 2]
```diff
@@ -135,6 +128,38 @@ which the driver has built-in shortcuts, so long as the relevant options
...
+analogous :ref:`table <table:mcscf_spaces>` for MCSCF is also available).  
+The third column of the
+table refers to CASCI, in which a full CI is performed in some smaller
+set of ``active`` orbitals; it is equivalent to CASSCF except without
+any orbital optimization.  It can be invoked via ``energy('fci')``
+with appropriate values selected for |globals__frozen_docc| and
+|globals__active|.  For CI computations, there is no difference between
+|globals__frozen_docc| and |globals__restricted_docc|, or between
+|globals__frozen_uocc| and |globals__restricted_uocc|.  There are
+differences betwen these keywords for :ref:`sec:mcscf`.
```
Comment (dgasmith) [inline doc/sphinxman/source/detci.rst]: Small typo here.

File: doc/sphinxman/source/detci.rst [hunk 3]
```diff
@@ -135,6 +128,38 @@ which the driver has built-in shortcuts, so long as the relevant options
...
+    +----------------------------+----------------------------+----------------------------+
+    | CI (e.g., CISD, FCI)       | RASCI                      | CASCI                      |
+    +============================+============================+============================+
+    | |globals__frozen_uocc|     | |globals__frozen_uocc|     |                            |
+    +----------------------------+----------------------------+                            +
+    | (all orbitals not in       | |globals__ras4|            |                            |
+    + |globals__frozen_uocc|     +----------------------------+                            +
+    | or |globals__frozen_docc|  | |globals__ras3|            |                            |
+    + are included in CI)        +----------------------------+                            +
+    |                            | |globals__ras2|            |                            |
```
Comment (loriab) [inline doc/sphinxman/source/detci.rst]: confirmed that this merged cell does not offend Sphinx

**Comment by CDSherrill** [comment]
Yes, @dgasmith, you're right, for completeness and consistency with the
other columns, I should add that.  I'll explain in the text how it can be
inferred from active.

**Comment by loriab** [comment]
For the users who suppose the table will tell them all they need and neglect to read the text, you can consider footnotes in the table. [Example here](https://raw.githubusercontent.com/psi4/psi4/master/doc/sphinxman/source/introduction.rst) (near bottom; search `f10`).

File: doc/sphinxman/source/detci.rst (hunk 2 above)
Comment (CDSherrill) [inline doc/sphinxman/source/detci.rst]: Working on this again today.... can't spot a typo in the above fragment... what's the typo?

Comment (jgonthier) [inline doc/sphinxman/source/detci.rst]: betwen instead of between

Comment (CDSherrill) [inline doc/sphinxman/source/detci.rst]: Thanks!!

**Pull request merged**